Hamilton_Matthew/detail.py

```diff
@@ -0,0 +1,49 @@
+from django import forms
+from django.shortcuts import render
+from django.http import HttpResponseRedirect
+from django.conf import settings
+from django_mako_plus import view_function, jscontext
+from formlib import Formless
+from django.utils import dateparse
+from datetime import datetime, timezone
+from catalog import models as cmod
+
+@view_function
+def process_request(request, product_id):
+
+    if not request.user.is_authenticated:
+        return HttpResponseRedirect("/account/login/")
+
+    utc_time = datetime.utcnow()
+    category_id = 0
+    page_title = "Details"
+
+    print(product_id)
+
+    #selects proper product
+    if cmod.Product.objects.filter(id = product_id).count() > 0:
+        product = cmod.Product.objects.get(id = product_id)
+        print(product.__class__.__name__)
+        #gets all product images
+        images = product.image_urls()
+        print(images)
+        category_id = product.category.id
+        page_title = product.name
+
```

Comment:
I suggest implementing a try catch here so errors can be shown if a non-valid product id is entered into the /detail url.

Reply:
That's what the if statement does